source/slang/check.cpp

```diff
@@ -9012,13 +8963,96 @@ namespace Slang
 
                 return createLookupResultExpr(
                     lookupResult,
-                    expr->BaseExpression,
+                    baseExpression,
                     expr->loc);
             }
             else if (as<ErrorType>(baseType))
             {
                 return CreateErrorExpr(expr);
             }
+
+            // Failure
+            return lookupMemberResultFailure(expr, baseType);
+        }
+
+        RefPtr<Expr> visitStaticMemberExpr(StaticMemberExpr* expr)
+        {
+            expr->BaseExpression = CheckExpr(expr->BaseExpression);
+
+            // Not sure this is needed -> but guess someone could do 
+            expr->BaseExpression = MaybeDereference(expr->BaseExpression);
+
+            // If the base of the member lookup has an interface type
+            // *without* a suitable this-type substitution, then we are
+            // trying to perform lookup on a value of existential type,
+            // and we should "open" the existential here so that we
+            // can expose its structure.
+            //
+
+            expr->BaseExpression = maybeOpenExistential(expr->BaseExpression);
```

Comment:
I think that both the `maybeDereference` and `maybeOpenExistential` here aren't needed, since we shouldn't be doing `foo::bar` when `foo` is an expression, and these two operations only apply to values (or pointer-like type and interface type, resepctively).